pettingzoo/test/api_test.py

```diff
@@ -383,6 +383,36 @@ def test_rewards_terminations_truncations(env, agent_0):
         test_reward(env.rewards[agent])
 
 
+def _test_observation_space_compatibility(expected, seen, context) -> None:
+    """Ensure observation's dtypes are same as in observation_space.
+
+    This tests that the dtypes of the spaces are the same.
+    The function will recursively check observation dicts to ensure that
+    all components have the same dtype as declared in the observation space.
+
+    Args:
+        expected: Observation space that is expected.
+        seen: The observation actually seen.
+        context: A list of all the dict keys that led to the current
```

Comment:
I don't quite understand this, what is context actually for? It doesn't seem to be used anywhere.

Reply:
Oh nvm I get it, it's recursion. I would prefer the term `recursed_keys`, but that's just a nit, I'll leave the decision to Elliot.

Reply:
I could see recursed keys or something else being more clear yeah

Reply:
Changed to recursed_keys. I'm all for clarity.